### ISSUE TITLE
How to modify files basing on answers (1.5.0) ?

### ISSUE THREAD

**Issue by CyprianSz** [saojs/sao#117]
I've got two questions and will appreciate help.

Is there any possibility to execute 'modify' action only on specific files ? When I'm trying to access 'answers' inside handler it's undefined.  Using 'when' or 'filters' doesn't seems to work either. Is there any way to handle such case ?

And second one is how to create value, which will be accessible by templating language but not getting it from answer ? It has to base on given ansewer but has to be modified and later use for templating. I also need original answer value so 'filter' is probably not an option.

Thanks in advance

**Comment by egoist** [comment]
> Is there any possibility to execute 'modify' action only on specific files

Yes, since you can access `this` (the [generator instance](https://saojs.org/generator-instance.html#properties)) in `actions`:

```js
module.exports = {
  actions() {
	return [{
      type: 'modify',
      files: this.answers.foo ? ['this-file'] : ['that-file'],
      handler: (data, filepath) => {}
    }]
  }
}
```

> And second one is how to create value, which will be accessible by templating language but not getting it from answer 

I just added an option [templateData](https://saojs.org/saofile.html#templatedata) to make this easier. (Available in 1.6.0)

**Comment by CyprianSz** [comment]
Great. Thank you for your answer.
